server/src/main/java/io/littlehorse/server/streams/storeinternals/ReadOnlyMetadataManager.java

```diff
@@ -24,20 +25,45 @@
 public class ReadOnlyMetadataManager {
 
     protected final Map<String, GetableToStore<?, ?>> uncommittedChanges = new TreeMap<>();
-    private final ReadOnlyModelStore defaultStore;
-    private final ReadOnlyModelStore tenantStore;
+    private final ReadOnlyClusterScopedStore clusterStore;
+    private final ReadOnlyTenantScopedStore tenantStore;
 
     public ReadOnlyMetadataManager(
-            final ReadOnlyModelDefaultStore defaultStore, final ReadOnlyTenantStore tenantStore) {
-        this.defaultStore = defaultStore;
-        this.tenantStore = tenantStore != null ? tenantStore : defaultStore;
+            final ReadOnlyClusterScopedStore clusterStore, final ReadOnlyTenantScopedStore tenantStore) {
+        this.clusterStore = clusterStore;
+        this.tenantStore = tenantStore;
     }
 
-    public <U extends Message, T extends GlobalGetable<U>> T get(MetadataId<?, U, T> id) {
+    public <U extends Message, T extends MetadataGetable<U>> T get(MetadataId<?, U, T> id) {
         log.trace("Getting {} with key {}", id.getType(), id);
         T out = null;
-        ReadOnlyModelStore specificStore = isClusterLevelObject(id) ? defaultStore : tenantStore;
+        // First check the cache.
+        @SuppressWarnings("unchecked")
```

Comment:
in some point we should remove this suppress right?

Reply:
IMO, `GetableToStore` shouldn't have a generic definition at class level. From what I understand, every class instance for this class only represent a row that will be store into the ModelStore, there is no point on setting a generic here because we are putting these instances in a buffer like this:

`Map<String, GetableToStore<?, ?>> uncommittedChanges.`

But then, you will be forced on casting objectToStore anyways. Unchecked casting are hard to avoid with this design, which tells me that we should think carefully every time that we define a generic class.

Reply:
This needs a ticket, also remove generic from `LHSerializable<>`